packages/tslint-config/src/rules/blueprintIconComponentsRule.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+import * as Lint from "tslint";
+import * as ts from "typescript";
+
+const OPTION_COMPONENT = "component";
+const OPTION_LITERAL = "literal";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    // tslint:disable:object-literal-sort-keys
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "blueprint-icon-components",
+        description: "Enforce usage of JSX Icon components over IconName string literals (or vice-versa)",
+        options: {
+            type: "array",
+            items: [{ enum: [OPTION_COMPONENT, OPTION_LITERAL], type: "string" }],
+            minLength: 1,
+            maxLength: 1,
+        },
+        optionsDescription: Lint.Utils.dedent`
+            One of the following two options must be provided:
+            * \`"${OPTION_COMPONENT}"\` requires JSX Icon components for \`icon\` props.
+            * \`"${OPTION_LITERAL}"\` requires \`IconName\` string literals for \`icon\` props.`,
+        optionExamples: [`[true, "${OPTION_COMPONENT}"]`, `[true, "${OPTION_LITERAL}"]`],
+        type: "functionality",
+        typescriptOnly: false,
+    };
+    // tslint:enable:object-literal-sort-keys
+
+    public static COMPONENT_MESSAGE = "use <NameIcon /> component for icon prop";
```

Comment:
sort of makes me think `<NameIcon />` is an actual component

Reply:
i might be able to improve the messaging here so it actually mentions icon name, but we gotta merge #2356 first. and i def want to add a fixer for this.

Reply:
well now it's `NamedIcon`